pkg/aws/eni/node.go

```diff
@@ -119,9 +120,17 @@ func (n *Node) PrepareIPRelease(excessIPs int, scopedLog *logrus.Entry) *ipam.Re
 	n.mutex.Lock()
 	defer n.mutex.Unlock()
 
+	// Needed for selecting the same ENI to release IPs from
+	// when more than one ENI qualifies for release
+	eniIds := make([]string, 0, len(n.enis))
+	for k := range n.enis {
+		eniIds = append(eniIds, k)
+	}
+	sort.Strings(eniIds)
```

Comment:
Same iterate order not ensure selecting the same ENI to release IPs in multi `PrepareIPRelease` called. what is use case ? @hemanthmalla

Reply:
Because select the ENI with the most addresses available for release  in every call `PrepareIPRelease`